### PR TITLE
feat: Jetpack Compose Navigation support

### LINKED ISSUE #2096: Jetpack Compose Navigation support
### Description

Add support for Jetpack Compose [Navigation library](https://developer.android.com/jetpack/compose/navigation)

- [x] Collect breadcrumbs for navigation/route changes, capturing from/to arguments as `data` (similar to [flutter](https://github.com/getsentry/sentry-dart/blob/main/flutter/lib/src/navigation/sentry_navigator_observer.dart)) #2116 and #2121 
- [x] Capture transactions for navigation/route changes (transactions should be idle) #2137 
- [x] Update [release registry](https://github.com/getsentry/sentry-release-registry) and .craft.yml 
	- [x] Craft https://github.com/getsentry/sentry-java/pull/2138
	- [x] Release registry https://github.com/getsentry/sentry-release-registry/pull/76 
- [ ] Update docs for:
	- [ ] New Navigation integration https://github.com/getsentry/sentry-docs/pull/5229
	- [ ] New Compose integration (android + desktop)
	- [ ] Migration docs (Kotlin bump to 1.6.10) https://github.com/getsentry/sentry-docs/pull/5219

### PR DESCRIPTION
**Pull request by romtsn**
## :scroll: Description

Integration branch to track compose navigation support

## :bulb: Motivation and Context

Closes #2096 

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

**Review by adinauer — APPROVED** [review]
Reviewed as separate PRs, so approving this as well

File: sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                if (fromArguments.isNotEmpty()) {
+                    data["from_arguments"] = fromArguments
+                }
+            }
+
+            val to = destination.route
+            to?.let { data["to"] = it }
+            arguments?.let { args ->
+                val toArguments = args.keySet().filter {
+                    it != NavController.KEY_DEEP_LINK_INTENT // there's a lot of unrelated stuff
```
Comment (philipphofmann) [inline sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt:51]: `m`: I think we miss a test case for this edge case here.

File: sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        val destination = mock<NavDestination>()
+        val navController = mock<NavController>()
+
+        fun getSut(toRoute: String = "route"): SentryNavigationListener {
+            whenever(destination.route).thenReturn(toRoute)
+            return SentryNavigationListener(hub)
+        }
+    }
+
+    private val fixture = Fixture()
```
Comment (philipphofmann) [inline sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt:36]: `l`: The fixture is a global state for all tests in this class. We set a permanent state to the `destination.route` with `whenever(destination.route).thenReturn(toRoute)`, which is not a problem at the moment because you only can create a sut by modifying this state, which might turn into a problem in the future resulting in stateful flaky tests. Maybe it's better to create a new fixture for every test?

File: sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryNavigationIntegration.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.navigation.NavController
+import androidx.navigation.NavHostController
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.android.navigation.SentryNavigationListener
+
+internal class SentryLifecycleObserver(
```
Comment (philipphofmann) [inline sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryNavigationIntegration.kt:16]: I guess this class is not finished yet?

**Review by philipphofmann — COMMENTED** [review]
What's missing to finish this PR?

File: sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryNavigationIntegration.kt (hunk 3 above)
Comment (romtsn) [reply sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryNavigationIntegration.kt:16]: Yes see #2137

File: sentry-android-navigation/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        // We run a full lint analysis as build part in CI, so skip vital checks for assemble tasks.
+        checkReleaseBuilds = false
+    }
+
+    variantFilter {
+        if (Config.Android.shouldSkipDebugVariant(buildType.name)) {
+            ignore = true
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (romtsn) [reply sentry-android-navigation/build.gradle.kts:56]: Yeah we should switch to convention plugins for that and configure everything there (in the buildSrc folder). Filed here https://github.com/getsentry/sentry-java/issues/2140

File: sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                }.associateWith { args[it] }
+                if (toArguments.isNotEmpty()) {
+                    data["to_arguments"] = toArguments
+                }
+            }
```
[the remark this answers is not in the document]
Comment (romtsn) [reply sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt:56]: already did in the other PR  that is gonna be merged in here

File: sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/compose/ComposeActivity.kt [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    navigateGithubWithArgs: () -> Unit
+) {
+    Column(
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier.fillMaxSize()
+    ) {
+        Button(
+            onClick = { navigateGithub() },
+            modifier = Modifier.padding(top = 32.dp)
```
[the remark this answers is not in the document]
Comment (romtsn) [reply sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/compose/ComposeActivity.kt:64]: mm, I'm not sure, maybe you can but since it's just a sample, I think we are good :)

File: sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt (hunk 1 above)
Comment (romtsn) [reply sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt:51]: that's right, but I think it's not crucial for the business logic - if this condition fails and we send this data, it won't hurt the user, right? They will just see some irrelevant stuff (or maybe it's actually relevant for them - we don't know, it was just an assumption for the most users). Though I can still add the test if you want me to.

Comment (philipphofmann) [reply sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt:51]: I would remove any code if it's not worth a test. It's up to you if you want to add a test.

File: sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt (hunk 2 above)
Comment (philipphofmann) [reply sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt:36]: Ah didn't know that. Maybe it was different in the past. Thanks for pointing that out.
